core/src/main/scala/akka/kafka/javadsl/Producer.scala

```diff
@@ -31,6 +33,20 @@ object Producer {
       .asJava
 
   /**
+   * The `plainSink` can be used for publishing records to Kafka topics.
+   * The `record` contains a topic name to which the record is being sent, an optional
+   * partition number, and an optional key and value.
+   */
+  def plainSink[K, V](
+    producerProvider: () => KafkaProducer[K, V],
```

Comment:
I want to unify both `ProducerSettings` and externally created producer API. In terms of stages I want to provide the API which allows creation of different `KafkaProducer` for each instances. That is why I kept `() => KafkaProducer` instead of just `KafkaProducer`.